tests/core/contracts/test_contract_call_interface.py

```diff
@@ -611,3 +614,23 @@ def test_invalid_fixed_value_reflections(web3, fixed_reflection_contract, functi
     contract_func = fixed_reflection_contract.functions[function]
     with pytest.raises(ValidationError, match=error):
         contract_func(value).call({'gas': 420000})
+
+
+def test_caller_default(math_contract):
+    result = math_contract.caller.return13()
+    assert result == 13
+
+
+def test_caller_with_parens(math_contract):
+    result = math_contract.caller().return13()
+    assert result == 13
+
+
+def test_caller_with_parens_and_transaction_dict(math_contract):
+    result = math_contract.caller({'from': 'notarealaddress.eth'}).add(2, 3)
```

Comment:
Another potentially interesting API would be to support sending these transaction fields in by keyword. `from` is problematic since it's a keyword, but we could go the pyethereum route and use `sender`.

Here that would mean:
```py
math_contract.caller(sender='notarealaddress.eth').add(2, 3)
```
which is kind of nice.

Reply:
Sounds good to me. So we would support both the `sender` kwarg and a transaction dict with `from` as a key, right?

Reply:
@carver I'd like to introduce that in a subsequent PR since it would make sense to make it available globally.

```python
web3.eth.sendTransaction(sender=..., gas=...)
contract.functions.transfer(...).transact(sender=...)
contract.caller(sender=...).add(...)
```